frontend/src/components/elements/Progress/Progress.test.tsx

```diff
@@ -41,17 +43,17 @@ describe("Progress Element", () => {
 
   it("moving backwards", () => {
     const props = getProps()
-    const wrapper = shallow(<Progress {...props} />)
-
-    expect(wrapper.instance().lastValue).toBe(50)
+    const wrapper = mount(<Progress {...props} />)
+    // To be checked for Functional component
+    // expect(wrapper.lastValue).toBe(50)
```

Comment:
I'll investigate into this

Reply:
This one is complex. If you use `useRef` instead, we'll want to do more. My solution involves mocking `useRef`

```
jest.mock('react', () => {
  const originReact = jest.requireActual('react');
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef,
  };
});
```

Annoyingly, since there are two refs, I found it easier hold both values.

```
const refs = useRef({
  lastValue: -1,
  lastAnimatedTime: -1
})
```

It makes then testing easier:

```
let mockRef;
  beforeEach(() => {
    mockRef = {
      current: {
        lastValue: -1,
        lastAnimatedTime: -1
      }
    }
    useRef.mockReturnValue(mockRef)
  })
```

Overall, it's a pain, but perhaps this might be a test util (patching `useRef`). It does get harder with multiple refs (since `useRef` is then called multiple times). It might be overkill. Curious what @kantuni thinks for this. Either we:
1. Do not test the ref
2. Do the patching as shown above (or something else if you have insight).
3. Mark this as not allowed to be a React Hook

Reply:
For the sake of simplicity, I recommend we do _not_ convert this to a hooked stateless component.

Reply:
Yes,reverting ProgressTracker